### DISCUSSION TITLE
Offer encrypted emails

### DISCUSSION THREAD

**Discussion by rixx** [pretix/pretix#1941] [Ideas]
Soo, bear with me.

I'd like for users to optionally include a PGP key/fingerprint when they enter their email address (if PGP is configured for the event address), and then receive all emails encrypted.

This would, of course, work best as a plugin, but it'd need to monkey patch the mail command, which isn't cool. So the real issue here is "provide mail hooks for plugins", I guess?

Edit: I'd be totally willing to build a plugin for that.

**Comment by raphaelm** [comment]
By "users", do you mean backend users or ticket buyers?

For backend users I'd even want to have this in core, as is it could be just another row in the user settings. It would then encrypt mails we send to them (which currently are only password resets and account change notifications but could be more in the future).

For ticket buyers this would need to go in a plugin.

**Comment by rixx** [comment]
I was only thinking about customers, hence the plugin. But having the functionality in core for backend users would of course make the plugin much simpler; I'm down for that, too.

**Comment by pc-coholic** [comment]
Since we have `pretix.base.signals.email_filter` now, we could totally have this live in a plugin... 

`pretix.base.signals.email_filter` checks if an order has been passed and if the order has the users key/fingerprint collected via `pretix.presale.signals.contact_form_fields` (order email) or `pretix.presale.signals.question_form_fields` (order position)
